bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/bridge/LongPolling.java

```diff
@@ -188,10 +193,28 @@ private void onLongPollComplete(BoschHttpClient httpClient, String subscriptionI
         if (failure != null) {
             handleLongPollFailure(subscriptionId, failure);
         } else {
-            logger.debug("Long poll response: {}", content);
+            handleLongPollResponse(httpClient, subscriptionId, content);
```

Comment:
One question: From the unit test `<TITLE>400</TITLE>` seems to suggest that actually the request failed with HTTP code 400? I'm wondering if `result.getResponse().getStatus()` would actually give you that error code?

In that case you could even check the code before trying to parse (in this case HTML) as JSON. Handling `JsonSyntaxException` is already a nice improvement, but there is no reason to try to parse it, if the request failed.

Reply:
I'm not 100% sure if there can be cases where we get a non-successful HTTP status code but still a valid JSON that we need to parse (for example, if a subscription is cancelled or other controlled long poll error conditions, see `org.openhab.binding.boschshc.internal.devices.bridge.dto.LongPollError`). I will add some debug logging to my local system and will try to find out more.

The issue to be fixed in this PR is unfortunately hard to reproduce because it only happens sporadically (approx. once every day, but not reliably every day on my system) so it will take a bit until I can confirm that the HTTP status code is the same as indicated in the HTML. Stay tuned.

Reply:
> I'm not 100% sure if there can be cases where we get a non-successful HTTP status code but still a valid JSON that we need to parse (for example, if a subscription is cancelled or other controlled long poll error conditions, see `org.openhab.binding.boschshc.internal.devices.bridge.dto.LongPollError`). I will add some debug logging to my local system and will try to find out more.

Good point, it might need some fine-grained control of which HTTP status codes to consider for such cases.

> The issue to be fixed in this PR is unfortunately hard to reproduce because it only happens sporadically (approx. once every day, but not reliably every day on my system) so it will take a bit until I can confirm that the HTTP status code is the same as indicated in the HTML. Stay tuned.

Sure. We can also merge as is and leave further improvements for a future PR. However, we are still a long way from first milestone, so we could wait and see your findings. Just let me know.